### DISCUSSION TITLE
build process

### DISCUSSION THREAD

**Discussion by master-hax** [shizunge/endlessh-go#143] [General]
i notice that we call `go mod tidy` in the [Dockerfile](https://github.com/shizunge/endlessh-go/blob/fb853625d1fcd227045de628fae31b3fae149cf1/Dockerfile) during the build: https://github.com/shizunge/endlessh-go/blob/fb853625d1fcd227045de628fae31b3fae149cf1/Dockerfile#L6 which updates `go.mod`

i'm curious why we need to do that - shouldn't the `go.mod` file be unchanged during the build?

**Comment by shizunge** [comment]
I think I added it because I am not always run the command when I checked in go.mod file.
Any suggestion? What is the drawback of adding it to the Dockerfile?

**Reply by master-hax** [reply]
i think it is generally a good idea to have deterministic builds guaranteed by the go.sum lock file.

doesn't `go mod tidy` re-calculate the `go.sum` lock file? this opens up the possibility of a supply chain attack that `go.sum` was supposed to protect us from at build time.

**Reply by master-hax** [reply]
essentially the golang equivalent of `npm install` bad, `npm ci` good
